File: 1-adhock/2-x-o/solution.java
class Main {

  private static boolean xo(String word) {
    int counter = 0;
    for (int i = 0; i < word.length(); i++) {
      char c = word.charAt(i);
      if (c == 'o') {
        counter += 1;
      } else if (c == 'x') {
        counter -= 1;
      }
    }
    if (counter == 0) {
      return true;
    } else {
      return false;
    }
  }

  public static void main(String[] args) {
    /** Driver Code */
    System.out.println(xo("xoxoxo")); // true
    System.out.println(xo("oxooxo")); // false
    System.out.println(xo("oxo")); // false
    System.out.println(xo("xxxooo")); // true
    System.out.println(xo("xoxooxxo")); // true
  }
}

File: 1-adhock/2-x-o/solution.py
def xo(str):
  counter = 0
  for i in str:
    if i == 'o':
      counter = counter + 1
    elif i == 'x':
      counter = counter - 1
  
  if counter == 0:
    return True
  else:
    return False

print(xo('xoxoxo')) # True
print(xo('oxooxo')) # False
print(xo('oxo')) # False
print(xo('xxxooo')) # True
print(xo('xoxooxxo')) # True